src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -654,21 +661,28 @@ private void DetermineTypeParameterMapping(ITypeSymbol inferredType, ITypeSymbol
                 ArgumentSyntax argumentOpt)
             {
                 var name = argumentOpt != null && argumentOpt.NameColon != null ? argumentOpt.NameColon.Name.Identifier.ValueText : null;
-                return InferTypeInArgument(index, parameterizedSymbols, name);
+                var refKind = RefKind.None;
+                if (argumentOpt != null && argumentOpt.RefOrOutKeyword != default(SyntaxToken))
+                {
+                    refKind = argumentOpt.RefOrOutKeyword.Kind() == SyntaxKind.OutKeyword ? RefKind.Out : RefKind.Ref;
+                }
+
+                return InferTypeInArgument(index, parameterizedSymbols, name, refKind);
             }
 
             private IEnumerable<ITypeSymbol> InferTypeInArgument(
                 int index,
                 IEnumerable<ImmutableArray<IParameterSymbol>> parameterizedSymbols,
-                string name)
+                string name,
+                RefKind refkind)
             {
                 // If the callsite has a named argument, then try to find a method overload that has a
                 // parameter with that name.  If we can find one, then return the type of that one.
                 if (name != null)
                 {
                     var parameters = parameterizedSymbols.SelectMany(m => m)
-                                                        .Where(p => p.Name == name)
-                                                        .Select(p => p.Type);
+                                                        .Where(p => p.Name == name && p.RefKind == refkind)
```

Comment:
We can call this with a null name or no refKind. I'm going to change it to:
* If there's a name, match parameters with the name
* If there's a refKind but no name, match the refkind

In the name + refKind case, it will only worry about matching the name.